### PR TITLE
Removed __div__ impls from Cython

### LINKED ISSUE #34704: BUG: CI builds failing at compilation of _sas
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

See, for example, the build line https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=37068&view=logs&j=a3a13ea8-7cf0-5bdb-71bb-6ac8830ae35c&t=2296ffae-5e67-52bf-66a8-6160145a0f80&l=660 
(and this appears to be failing other PRs too)

#### Problem description

CI builds are failing, see link above

### PR DESCRIPTION
**Pull request by WillAyd**
Alternate to #34711 which may close #34704

### CODE REVIEW

**Comment by WillAyd** [comment]
I believe failure is unrelated. Looking into it but @jbrockmendel might have some ideas as well given his comment here ref #33858:

https://github.com/pandas-dev/pandas/blob/08febd2b9f77fd8f943de92939788e8bc53ee2d2/pandas/tests/extension/base/dtype.py#L78

**Comment by WillAyd** [comment]
Nevermind had the wrong PR. Looks like it _might_ trace back to #22699

**Comment by jreback** [comment]
hmm what is this needed for?

**Comment by WillAyd** [comment]
These throw unused function warnings / errors in newer Cython versions during compilation

File: ci/deps/azure-37-locale.yaml [hunk 1]
```diff
@@ -5,8 +5,7 @@ dependencies:
   - python=3.7.*
 
   # tools
-  # Cython pin for https://github.com/pandas-dev/pandas/issues/34704
```
Comment (jreback) [inline ci/deps/azure-37-locale.yaml:8]: do we want to revert this pin?

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/deps/azure-37-locale.yaml (hunk 1 above)
Comment (WillAyd) [reply ci/deps/azure-37-locale.yaml:8]: Yea that pin was a quick fix for CI but not needed if you remove these div methods

Comment (jreback) [reply ci/deps/azure-37-locale.yaml:8]: i c ok

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Comment by simonjayhawkins** [comment]
@MeeseeksDev backport to 1.0.x

**Comment by simonjayhawkins** [comment]
xref #34802

**Pull request merged**